### ISSUE TITLE
Comment about glFlushRenderAPPLE

### ISSUE THREAD

**Issue by vade** [v002/v002-Open-Kinect#3]
hey @bangnoise - I'm doing some tweaking to the Kinect Patch - on Nvidia hardware (I've yet to test ATI or Intel) disabling this causes massive stuttering / flickering in the image.

I think this is because that the textures are using client storage and texture range, but the kinect is swapping the buffers out from under the texture backing. Because QC is lazily evaluated, and texture submission happens open the texture being requested, the buffer is transiting / in flight.

the glFlushRenderAPPLE appears to solve that - but probably a lock is better, or a queue of buffers perhaps.

Just throwing it out there if you have a better idea, insight or what not.

Thanks!

**Comment by bangnoise** [comment]
Yep those extensions require the backing remains valid for the lifetime of the texture - if it changes during drawing then you will get tearing or unexpected content.

If you want to avoid the flush but still use the extensions then - assuming a flush is happening at some point between render passes - I'd probably have two buffers and use an alternate buffer each pass. If you want a thread to wait for a buffer to be drawn before updating it, then I guess a GL fence.
